cmd/build/v2/smartbuild/hasher.go

```diff
@@ -110,36 +117,22 @@ func (sh serviceHasher) hash(buildInfo *model.BuildInfo, commitHash string) stri
 	fmt.Fprintf(&b, "secrets:%s;", secretsText)
 	fmt.Fprintf(&b, "context:%s;", buildInfo.Context)
 	fmt.Fprintf(&b, "dockerfile:%s;", buildInfo.Dockerfile)
-	fmt.Fprintf(&b, "dockerfile_content:%s;", getDockerfileContent(buildInfo.Context, buildInfo.Dockerfile))
+	fmt.Fprintf(&b, "dockerfile_content:%s;", sh.getDockerfileContent(buildInfo.Context, buildInfo.Dockerfile))
+	fmt.Fprintf(&b, "diff:%s;", diff)
```

Comment:
why include always `diff` ?. I wouldn't include what is not needed in the hash. For project commit hash is always emtpy.

Reply:
It's empty but it simplifies the code by not needing to have two different strategies. It also doesn't take time to compute it.